### PR TITLE
Populate the ValidIssuer and the ValidAudience properties of the JwtBearerOptions.TokenValidationParameters.

### PR DESCRIPTION
**Pull request by satma0745**
PR for the [Issue #52820](https://github.com/dotnet/aspnetcore/issues/52820).

### CODE REVIEW

**Comment by martincostello** [comment]
Ideally this PR should include 1 or more tests that validate the fix and that it doesn't get broken again in the future by accident.

**Comment by satma0745** [comment]
@dotnet-policy-service agree

**Comment by satma0745** [comment]
@martincostello, I've refactored old tests a bit and provide my own one.

Now we have 3 tests related to reading TokenValidationParameters from configuration:
1. [CanReadJwtBearerOptionsFromConfig](https://github.com/dotnet/aspnetcore/blob/96101d4853dfb9230ac7d7de911207a6bb1924bd/src/Security/Authentication/test/JwtBearerTests.cs#L1014) tests the base case for simple flat values
2. [CanReadMultipleAudiencesFromConfig](https://github.com/dotnet/aspnetcore/blob/96101d4853dfb9230ac7d7de911207a6bb1924bd/src/Security/Authentication/test/JwtBearerTests.cs#L1049C46-L1049C46) tests if the array of ValidAudiences is loaded properly
3. [CanReadMultipleIssuersFromConfig](https://github.com/dotnet/aspnetcore/blob/96101d4853dfb9230ac7d7de911207a6bb1924bd/src/Security/Authentication/test/JwtBearerTests.cs#L1074C30-L1074C30) tests that ValueIssuers and IssuerSigningKeys are loaded and threated properly. I think it's better to have it in a separate test-method since they have complex rules (in contrast to the ValidAudiences property).

**Comment by satma0745** [comment]
I tried sticking with the established convention of testing and designed the tests not to stand out too much.
Anyway, let me know if You find something looking out of place or little bit off.

**Comment by martincostello** [comment]
> or by simply closing and reopening

**Comment by satma0745** [comment]
> > or by simply closing and reopening

But what's about the Assigned Developer? Won't he be un-assigned when I close the PR?

**Comment by martincostello** [comment]
Not to my knowledge.

File: src/Security/Authentication/test/JwtBearerTests.cs [hunk 1]
```diff
@@ -1017,8 +1017,7 @@ public void CanReadJwtBearerOptionsFromConfig()
         var config = new ConfigurationBuilder().AddInMemoryCollection(new[]
         {
             new KeyValuePair<string, string>("Authentication:Schemes:Bearer:ValidIssuer", "dotnet-user-jwts"),
-            new KeyValuePair<string, string>("Authentication:Schemes:Bearer:ValidAudiences:0", "http://localhost:5000"),
-            new KeyValuePair<string, string>("Authentication:Schemes:Bearer:ValidAudiences:1", "https://localhost:5001"),
+            new KeyValuePair<string, string>("Authentication:Schemes:Bearer:ValidAudience", "http://localhost:5000"),
```
Comment (halter73) [inline src/Security/Authentication/test/JwtBearerTests.cs:1020]: Can we include a test that configures both `ValidIssuer` and `ValidIssuers`? Or `ValidAudience` and `ValidAudiences`?

**Review by halter73 — COMMENTED** [review]
@captainsafia Can you take a look at this since you worked on this and very clearly intentionally set just the multi-value properties?

> I noticed while exploring the identity model code that even though they expose single-value properties for config (e.g. `ValidIssuer` and `ValidAudience`) it looks like the multi-value properties are ultimately used as the source of truth. Updated this to just set the multi-value property.

https://github.com/dotnet/aspnetcore/pull/42679#discussion_r922378053

I think the current behavior of not setting the single-value properties is fine since it gets included in the multi-value property. And based on my reading of the [ValidateIssuerAsync](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/55db56e506f0d7466e97ed45604c05166b322dea/src/Microsoft.IdentityModel.Tokens/Validators.cs#L288-L314) method in IdentityModel, this change is redundant since it checks against `ValidIssuer` and then all the `ValidIssuers`.

If we decide to change the config binding at all, we should make it align with the `TokenValidationParameters` more directly and not include the single-value property in the multi-value property. That might make things less confusing, and I don't think it would cause any real problems since it appears IdentityModel looks at both together.

**Review by halter73 — APPROVED** [review]
I talked to @captainsafia, and we're fine taking this change since we have quite a few previews left to see if there are any problems with this simplification.

**Pull request merged**